### PR TITLE
[SPARK-38796][SQL][DOC][FOLLOWUP] Remove try_to_char reference in the doc

### PR DESCRIPTION
**Pull request by cloud-fan**
### What changes were proposed in this pull request?

We have removed the `try_to_char` function and it shouldn't appear in the doc anymore. This PR also improves the doc a little bit.

### Why are the changes needed?

fix doc

### Does this PR introduce _any_ user-facing change?

no

### How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
CC @HyukjinKwon

File: docs/sql-ref-number-pattern.md [hunk 1]
```diff
@@ -176,10 +176,10 @@ Note that the format string used in most of these examples expects:
   "$#.##"
 
 -- 'S' can be at the end.
-> SELECT try_to_char(decimal(-12454.8), '99,999.9S');
+> SELECT to_char(decimal(-12454.8), '99,999.9S');
   "12,454.8-"
 
-> SELECT try_to_char(decimal(12454.8), 'L99,999.9');
+> SELECT to_char(decimal(12454.8), 'L99,999.9');
   Error: cannot resolve 'try_to_char(Decimal(12454.8), 'L99,999.9')' due to data type mismatch:
```
Comment (dtenedor) [inline docs/sql-ref-number-pattern.md:183]:
```suggestion
  Error: cannot resolve 'to_char(Decimal(12454.8), 'L99,999.9')' due to data type mismatch:
```

**Review by dtenedor — APPROVED** [review]
Thanks for improving the docs!

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**